internal/headers/access_token.go

```diff
@@ -0,0 +1,62 @@
+package headers
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+
+	"github.com/coreos/go-oidc"
+	"github.com/rs/zerolog"
+)
+
+type tokenResponse struct {
+	AccessToken string `json:"access_token"`
+	Scope       string `json:"scope"`
+}
+
+const (
+	scopes     = "openid api.iam.service_accounts"
+	grant_type = "client_credentials"
+)
+
+// https://sso.redhat.com/auth/realms/redhat-external/protocol/openid-connect/token" -d "scope=openid api.iam.service_accounts
+func getToken(ctx context.Context, issuerUrl, clientId, clientSecret string) (string, error) {
+	provider, err := oidc.NewProvider(ctx, issuerUrl)
+	if err != nil {
+		return "", err
+	}
+
+	data := url.Values{}
+	data.Add("grant_type", grant_type)
+	data.Add("scope", scopes)
+	data.Add("client_id", clientId)
+	data.Add("client_secret", clientSecret)
+
+	req, err := http.NewRequest(http.MethodPost, provider.Endpoint().TokenURL, strings.NewReader(data.Encode()))
```

Comment:
This is me being bit fancy just to resolve a static URL, for which we need extra module :thinking: 

provider.Endpoint().TokenURL is quite static value.